lm_eval/api/task.py

```diff
@@ -41,6 +41,22 @@
 eval_logger = logging.getLogger("lm-eval")
 
 
+@dataclass
+class GroupConfig(dict):
+    group: str = None
+    task: Union[str, list] = None
+    weight_by_size: bool = False
```

Comment:
after we merge this PR, I'd like to add a `metadata` field to GroupConfigs in order to close #1360 .

regarding versioning, not sure yet what our rule for groups' versioning should be though.

Reply:
Oh okay in that case Let's keep `GroupConfig`. I didn't end up using it.

Reply:
Oh if it's not yet used, then we could add GroupConfig later.

I think it'd probably be nice to have explicitly, but trust your judgement more on what's necessary